### ISSUE TITLE
Why are `@types` dependencies not listed as `devDependencies`?

### ISSUE THREAD

**Issue by koddsson** [airbnb/visx#1114]
👋🏻 

I'm getting these packages in my app even though I presumably don't need these types at all?

https://github.com/airbnb/visx/blob/6f5391724f8d48f24f953ed45b74080e0ad28b02/packages/visx-responsive/package.json#L31-L32

I could be missing something but should these be installed at devDependencies?

**Comment by williaster** [comment]
Hi @koddsson 👋 . In libraries (**not apps**), types usually go in `dependencies` and not `devDependencies`. 

If we use or reference a type (say from `d3`) in `visx` and placed that type dependency in `devDependencies`, and then export a component that uses the type, or export a `type` that extends that `d3` type, consumers of `visx` would not have access to it because `visx` `devDependencies` are _not_ installed when you install a `visx` library. There might be some libraries that are purely internal to `visx` (perhaps `lodash`) but to err on the side of caution and not causing type issues for consumers we have put them in `dependencies.

Here's a similar [`stack overflow question`](https://stackoverflow.com/a/46011417/1933266).

**Comment by keithamus** [comment]
Hey @williaster!

The bundle should not output typescript, and should not rely on references from `@type` packages. Bundles should be outputting JavaScript, with `.d.ts` sidecar files. The upstream project may or may not be using TS, but if it is (in the case of `github/github` it is) then it is up to that project to resolve tooling conflicts, and that's what we'll do upstream - by installing any missing `@types/` libraries.

The TL;DR is that the `@types/` packages all most definitely belong in `devDependencies`.
